service_taxi/src/main/kotlin/org/thechance/service_taxi/data/gateway/DataBaseGatewayImpl.kt

```diff
@@ -0,0 +1,126 @@
+package org.thechance.service_taxi.data.gateway
+
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.Updates
+import org.bson.types.ObjectId
+import org.koin.core.annotation.Single
+import org.litote.kmongo.and
+import org.litote.kmongo.eq
+import org.litote.kmongo.ne
+import org.thechance.service_taxi.api.models.taxi.toCollection
+import org.thechance.service_taxi.api.models.taxi.toEntity
+import org.thechance.service_taxi.api.models.trip.toCollection
+import org.thechance.service_taxi.api.models.trip.toEntity
+import org.thechance.service_taxi.data.DataBaseContainer
+import org.thechance.service_taxi.data.collection.TaxiCollection
+import org.thechance.service_taxi.data.collection.TripCollection
+import org.thechance.service_taxi.data.utils.isNotNull
+import org.thechance.service_taxi.data.utils.isSuccessfullyUpdated
+import org.thechance.service_taxi.data.utils.paginate
+import org.thechance.service_taxi.data.utils.updateNotNullProperties
+import org.thechance.service_taxi.domain.entity.Taxi
+import org.thechance.service_taxi.domain.entity.TaxiUpdateRequest
+import org.thechance.service_taxi.domain.entity.Trip
+import org.thechance.service_taxi.domain.entity.TripUpdateRequest
+import org.thechance.service_taxi.domain.gateway.DataBaseGateway
+
+@Single
+class DataBaseGatewayImpl(private val container: DataBaseContainer) : DataBaseGateway {
+    // region taxi curd
+    override suspend fun addTaxi(taxi: Taxi): Boolean {
+        return container.taxiCollection.insertOne(taxi.toCollection()).wasAcknowledged()
+    }
+
+    override suspend fun getTaxiById(taxiId: String): Taxi? {
+        return container.taxiCollection.findOneById(ObjectId(taxiId))
+            ?.takeIf { !it.isDeleted }?.toEntity()
```

Comment:
In your case, it will retrieve ALL the data(is deleted included), then it will filter it in the server side here.. you can filter it in the mongodb directly

Reply:
but it's not filters, it's simple if condition @Mohammed-Alhams